config/helm/networkpolicy_capa-controller-manager.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  labels:
+    app.kubernetes.io/component: cluster-api
+    cluster.x-k8s.io/provider: infrastructure-aws
+    control-plane: capa-controller-manager
+  name: capa-controller-manager
+spec:
+  podSelector:
+    matchLabels:
+      cluster.x-k8s.io/provider: infrastructure-aws
+      control-plane: capa-controller-manager
+  egress:
+    # Kubernetes API on workload cluster (typically via load balancer), AWS API
```

Comment:
should we add a comment explaining why this case can't be covered by `CiliumNetworkPolicies`?

Reply:
I'm using the standard `NetworkPolicy` where no CNI-specific manifest is needed. Does that make enough sense as explanation? Same happened for our CAPI app.

Reply:
Trying to get some discussion going: https://gigantic.slack.com/archives/CLPMFRVU6/p1690796652450279

Reply:
We didn't really get to much of a conclusion, but I think it's fair to use `NetworkPolicy` wherever it's sufficient (as we also do in `cluster-api-app`) and `CiliumNetworkPolicy` only where needed. This avoids too much vendor lock-in.